pkg/server/api/enroll.go

```diff
@@ -124,7 +153,18 @@ func proxyAuthenticated(ap *config.AuthenticatingProxy, w http.ResponseWriter, r
 		return "", false
 	}
 
-	if !(clientAuthenticated(r) && clientHostnameMatches(ap.Hostname, r)) {
+	// Host name matching
+	hostnameMatches := (clientAuthenticated(r) && clientHostnameMatches(ap.Hostname, r))
+
+	// SPIFFE ID matching
+	spiffeIdMatches, err := clientSpiffeIdMatches(ap.SpiffeId, r)
+	if err != nil {
+		// Do not necessarily fail as other conditions might apply
+		logger.Error("malformed proxy spiffe id")
+	}
+
+	// Matching fails case
+	if !(hostnameMatches || spiffeIdMatches) {
 		logHttpError(r, w, fmt.Errorf("request didn't come from proxy"), http.StatusUnauthorized, logger)
```

Comment:
It would be nice to know what hostname/SPIFFE ID was on the client cert (if there was a client cert) to aid with debugging. It's an existing problem, though.